src/Classes/SkillListControl.lua

```diff
@@ -178,3 +195,30 @@ function SkillListClass:OnHoverKeyUp(key)
 		end
 	end
 end
+
+
+function SkillListClass:Draw(viewPort)
+	self.ListControl.Draw(self, viewPort)
+end
+
+function SkillListClass:GetRowIcon(column, index, socketGroup)
+	if column == 1 then
+		local slot = socketGroup.slot or nil
+		local color = "^7"
+		local currentMainSkill = self.skillsTab.build.mainSocketGroup == index
+		local disabled = not socketGroup.enabled or not socketGroup.slotEnabled
+		if disabled then
```

Comment:
Here's a partial implementation of changing the icon depending on the item type. I strongly suggest keeping the weapon icon consistent across all weapon types, but differentiating between an off-hand weapon, a shield, and a quiver. 

You'd also need to add icons for a quiver, and possibly an off hand weapon (maybe just invert the main weapon icon on the Y axis or something?)

```suggestion
local itemsTab = self.skillsTab.build.itemsTab
		local weapon2Sel = itemsTab.activeItemSet["Weapon 2"].selItemId or 0
		local weapon2Type = itemsTab.items[weapon2Sel] and itemsTab.items[weapon2Sel].base.type or "None"

		if slot == "Weapon 2" and (weapon2Type == "Quiver" or weapon2Type == "Shield") then
			slot = weapon2Type
		elseif slot == "Weapon 2" then
			slot = "Weapon 1"
		end
		if disabled then
```